### ISSUE TITLE
[BUG] php layer error while installing php-extras

### ISSUE THREAD

**Issue by crinklywrappr** [syl20bnr/spacemacs#10566]
C-c C-c didn't work :/

#### Description :octocat:

Receiving this error when loading the php layer at startup:
`An error occurred while installing php-extras (error: (wrong-type-argument stringp nil))`

#### Reproduction guide :beetle:

- Start Emacs

- Add `php` layer.

- `SPC f e r` or re-open

*Observed behaviour:* :eyes: :broken_heart:

It doesn't work

*Expected behaviour:* :heart: :smile:

It should work

#### System Info :computer:
- OS: windows-nt
- Emacs: 25.3.1
- Spacemacs: 0.200.13
- Spacemacs branch: master (rev. 582f9aa4)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(csv javascript python html sql php helm emacs-lisp auto-completion markdown clojure
     (shell :variables shell-default-shell 'eshell shell-enable-smart-eshell t shell-default-height 30 shell-default-position 'bottom))
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES

**Comment by duianto** [comment]
TLDR: It seems like you need to install the `tar` executable.

This might be a duplicate of this issue:
Php-extras error: wrong-type arguement stringp nil #5007 

The `php-extras` author commented there, stating that the issue might be related to a missing `tar` executable.

That issue linked to this issue:
WINDOWS-10 An error occurred while installing php-extras #7517

where the last comment states that it was solved by reading this issue:
Windows php-extras in spacemacs by quelpa https://github.com/quelpa/quelpa/issues/118

There they discuss installing `tar` in windows.

There seems to be two ways of installing `tar`:
https://github.com/quelpa/quelpa#tar
and
https://github.com/quelpa/quelpa#cygwin

**Comment by sdwolfz** [comment]
I've noticed this problem on my work mabook and excluded `php-extras` to get around it. Will let you know on monday if installing `tar` solves the problem.

**Comment by doubleagent** [comment]
Installing `tar` and adding it to the path didn't solve the problem for me.  I'm seeing the same error message.

**Comment by sdwolfz** [comment]
I can confirm that installing `gnu-tar` on OSX fixes this problem. I had `bsdtar` before but it does not work with it.

**Comment by Azd325** [comment]
> I can confirm that installing `gnu-tar` on OSX fixes this problem. I had `bsdtar` before but it does not work with it.
> 
> ```shell
> brew install gnu-tar --with-default-names
> ```

Did not work for me.

**Comment by zendo** [comment]
install git for windows, than add this to your config. `(setq-default quelpa-build-tar-executable "C:/Program Files/Git/usr/bin/tar")`

**Comment by duianto** [comment]
> install git for windows, than add this to your config. `(setq-default quelpa-build-tar-executable "C:/Program Files/Git/usr/bin/tar")`

Confirmed.
Note that it needs to be added to the `dotspacemacs/user-init` section, it didn't work from the `dotspacemacs/user-config` section of `.spacemacs`.

#### System Info :computer:
- OS: windows-nt
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 3d39b98e9)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey auto-completion command-log emacs-lisp git haskell helm imenu-list markdown multiple-cursors
            (org :variables org-agenda-files
                 '("c:/Users/username/org/notes.org"))
            php
            (shell :variables shell-default-shell 'eshell)
            spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2

**Comment by duianto** [comment]
The workaround worked when my previous comment was posted https://github.com/syl20bnr/spacemacs/issues/10566#issuecomment-575889140

I don't know what's changed, but I'm getting the error again:
>An error occurred while installing php-extras (error: (wrong-type-argument package-desc nil))

<details><summary>
php-extras related entries in the messages buffer
</summary>
<pre>
(Spacemacs) --> installing package: php-extras@php... [1/1]
Fetcher: github
Source: arnested/php-extras

Updating c:/Users/username/.emacs.d/.cache/quelpa/build/php-extras/
c:/Users/username/.emacs.d/.cache/quelpa/build/php-extras/php-extras-gen-eldoc.el -> c:/Users/username/AppData/Local/Temp/php-extrasA2GdL6/php-extras-20160518.1716/php-extras-gen-eldoc.el
c:/Users/username/.emacs.d/.cache/quelpa/build/php-extras/php-extras.el -> c:/Users/username/AppData/Local/Temp/php-extrasA2GdL6/php-extras-20160518.1716/php-extras.el
c:/Users/username/.emacs.d/.cache/quelpa/build/php-extras/dir -> c:/Users/username/AppData/Local/Temp/php-extrasA2GdL6/php-extras-20160518.1716/dir
Wrote c:/Users/username/.emacs.d/.cache/quelpa/packages/php-extras-readme.txt
Error getting PACKAGE-DESC: (file-missing Opening input file No such file or directory c:/Users/username/.emacs.d/.cache/quelpa/packages/php-extras-20160518.1716.tar)
(Spacemacs) Error: 
An error occurred while installing php-extras (error: (wrong-type-argument package-desc nil))

</pre>
</details>

The workaround:
```
  (setq-default quelpa-build-tar-executable "C:/Program Files/Git/usr/bin/tar")
```
doesn't work from either the user-init or user-config sections.

There are 3 tar versions installed:
```
λ where tar
C:\tools\cygwin\bin\tar.exe
C:\Windows\System32\tar.exe
C:\Program Files\Git\usr\bin\tar.exe
```
But the following command:
```
(executable-find "tar")
```
only finds:
>"c:/tools/cygwin/bin/tar.exe"

instead of the expected git for windows `tar.exe`

I added the path to the directory that contains the git: `tar.exe`
```
C:\Program Files\Git\usr\bin\
```
to the beginning of the Windows Environment Variable: `PATH`
and to the `.spacemacs.env` line: `PATH`

But it still doesn't work.

**Comment by ajohnston1219** [comment]
> The workaround worked when my previous comment was posted [#10566 (comment)](https://github.com/syl20bnr/spacemacs/issues/10566#issuecomment-575889140)
> 
> I don't know what's changed, but I'm getting the error again:
> 
> > An error occurred while installing php-extras (error: (wrong-type-argument package-desc nil))
> 
>  php-extras related entries in the messages buffer
> The workaround:
> 
> ```
>   (setq-default quelpa-build-tar-executable "C:/Program Files/Git/usr/bin/tar")
> ```
> 
> doesn't work from either the user-init or user-config sections.
> 
> There are 3 tar versions installed:
> 
> ```
> λ where tar
> C:\tools\cygwin\bin\tar.exe
> C:\Windows\System32\tar.exe
> C:\Program Files\Git\usr\bin\tar.exe
> ```
> 
> But the following command:
> 
> ```
> (executable-find "tar")
> ```
> 
> only finds:
> 
> > "c:/tools/cygwin/bin/tar.exe"
> 
> instead of the expected git for windows `tar.exe`
> 
> I added the path to the directory that contains the git: `tar.exe`
> 
> ```
> C:\Program Files\Git\usr\bin\
> ```
> 
> to the beginning of the Windows Environment Variable: `PATH`
> and to the `.spacemacs.env` line: `PATH`
> 
> But it still doesn't work.
> 
>  Windows 1903

I am having this same issue with Windows 1909

**Comment by zilti** [comment]
I am having this same issue on *Linux*, so tar is definitely not an issue
